src/Microsoft.OData.Core/ODataWriterCore.cs

```diff
@@ -903,13 +1059,13 @@ protected virtual void WritePrimitiveValue(ODataPrimitiveValue primitiveValue)
         /// </summary>
         /// <param name="resourceSet">The resource set for the new scope.</param>
         /// <param name="navigationSource">The navigation source we are going to write resource set for.</param>
-        /// <param name="resourceType">The structured type for the items in the resource set to be written (or null if the entity set base type should be used).</param>
+        /// <param name="itemType">The structured type for the items in the resource set to be written (or null if the entity set base type should be used).</param>
         /// <param name="skipWriting">true if the content of the scope to create should not be written.</param>
         /// <param name="selectedProperties">The selected properties of this scope.</param>
         /// <param name="odataUri">The ODataUri info of this scope.</param>
         /// <param name="isUndeclared">true if the resource set is for an undeclared property</param>
         /// <returns>The newly create scope.</returns>
-        protected abstract ResourceSetScope CreateResourceSetScope(ODataResourceSet resourceSet, IEdmNavigationSource navigationSource, IEdmStructuredType resourceType, bool skipWriting, SelectedPropertiesNode selectedProperties, ODataUri odataUri, bool isUndeclared);
+        protected abstract ResourceSetScope CreateResourceSetScope(ODataResourceSet resourceSet, IEdmNavigationSource navigationSource, IEdmType itemType, bool skipWriting, SelectedPropertiesNode selectedProperties, ODataUri odataUri, bool isUndeclared);

```

Comment:
>IEdmType itemType [](start = 136, length = 17)

To CreateResourceSetScope using "IEdmType" looks not sense.

Reply:
ResourceSet is used for any collection, including (now) collections of primitives that are written individually.  Exchange requirement that not all (primitive) elements of a collection would be written at once.

---
In reply to: [268361587](https://github.com/OData/odata.net/pull/1204#discussion_r268361587) [](ancestors = 268361587)